cloud/google/gce.py

```diff
@@ -308,15 +308,17 @@ def create_instances(module, gce, instance_names):
     # with:
     # [ {'key': key1, 'value': value1}, {'key': key2, 'value': value2}, ...]
     if metadata:
-        try:
-            md = literal_eval(str(metadata))
-            if not isinstance(md, dict):
-                raise ValueError('metadata must be a dict')
-        except ValueError, e:
-            module.fail_json(msg='bad metadata: %s' % str(e))
-        except SyntaxError, e:
-            module.fail_json(msg='bad metadata syntax')
-
+        if isinstance(metadata, dict):
+            md = metadata
```

Comment:
If I understand GCE this needs to be a dict with keys and values as strings. Do we want to enforce that level of 'schema'?